playbooks/archive_artifacts.yml

```diff
@@ -0,0 +1,77 @@
+- hosts: "{{ hosts }}"
+  tasks:
+    - name: Create temporary directory to store artifacts
+      file:
+        path: "{{ artifacts_path }}"
+        state: directory
+      delegate_to: "localhost"
+      run_once: yes
+
+    - name: Grab host data
+      command: >
+               rsync
+               --archive
+               --compress
+               --verbose
+               --relative
+               --rsh 'ssh -o StrictHostKeyChecking=no'
+               --ignore-missing-args
+               {{ inventory_hostname }}:{{ item }}
+               {{ artifacts_path }}/{{ ansible_hostname }}
+      with_items:
+        - "/openstack/log/{{ ansible_hostname }}-lxc"
+        - "/etc"
+        - "/var/log"
+      delegate_to: "localhost"
+
+    - name: Grab leapfrog upgrade .leap marker files
+      command: >
+               rsync
+               --archive
+               --compress
+               --verbose
+               --rsh 'ssh -o StrictHostKeyChecking=no'
+               --include '*.leap'
+               --exclude '*'
+               {{ inventory_hostname }}:/opt/rpc-leapfrog/leap42/
+               {{ artifacts_path }}/{{ ansible_hostname }}/leap_marker_files
+      delegate_to: "localhost"
+      ignore_errors: yes
+
+    - name: List containers
+      command: "lxc-ls -1"
+      failed_when:
+        - "{{ containers.rc != 0 }}"
+        - "{{ containers.msg != '[Errno 2] No such file or directory' }}"
```

Comment:
If "this" refers to ignoring missing directories, it is acceptable because it means `lxc-ls` isn't installed and not all hosts have it e.g. nova-compute and deploy hosts.